src/acceleration/BaseQNAcceleration.cpp

```diff
@@ -275,18 +286,19 @@ void BaseQNAcceleration::updateDifferenceMatrices(
 void BaseQNAcceleration::performAcceleration(
     DataMap &cplData)
 {
-  PRECICE_TRACE(_dataIDs.size(), cplData.size());
+  PRECICE_TRACE(_primaryDataIDs.size(), cplData.size());
 
   profiling::Event e("cpl.computeQuasiNewtonUpdate", profiling::Synchronize);
 
-  PRECICE_ASSERT(_oldResiduals.size() == _oldXTilde.size(), _oldResiduals.size(), _oldXTilde.size());
-  PRECICE_ASSERT(_values.size() == _oldXTilde.size(), _values.size(), _oldXTilde.size());
-  PRECICE_ASSERT(_oldValues.size() == _oldXTilde.size(), _oldValues.size(), _oldXTilde.size());
-  PRECICE_ASSERT(_residuals.size() == _oldXTilde.size(), _residuals.size(), _oldXTilde.size());
+  PRECICE_ASSERT(_oldPrimaryResiduals.size() == _oldPrimaryXTilde.size(), _oldPrimaryResiduals.size(), _oldPrimaryXTilde.size());
+  PRECICE_ASSERT(_primaryValues.size() == _oldPrimaryXTilde.size(), _primaryValues.size(), _oldPrimaryXTilde.size());
+  PRECICE_ASSERT(_oldPrimaryValues.size() == _oldPrimaryXTilde.size(), _oldPrimaryValues.size(), _oldPrimaryXTilde.size());
+  PRECICE_ASSERT(_primaryResiduals.size() == _oldPrimaryXTilde.size(), _primaryResiduals.size(), _oldPrimaryXTilde.size());
 
   // assume data structures associated with the LS system can be updated easily.
 
   // scale data values (and secondary data values)
```

Comment:
Is this comment correct? I don't think that there is any scaling in there.

Reply:
Only concatenate happens until the next comment

Reply:
But then the comment is misleading, no?

Reply:
now, it says only `scale data values`, but there is still no scaling happening, no? :)